code-challenges.test.js

```diff
@@ -12,41 +12,169 @@
 
 // --------------------1) Create a function that takes in a string and returns a string with a coded message. The coded message converts "a" to 4, "e" to 3, "i" to 1, and "o" to 0.
 
+// Pseudocode
+// Input: String 
+// Output: String with coded message
+// Function name: codedMessage
+// Process: replace letters in the input string using the .replace method to replace all instances of the letters "a", "e", "i", and "o" (both uppercase and lowercase using regex and the case insensitive property and global property /ig) in the input string with the numbers 4, 3, 1, and 0 and return the modified string 
+
 // a) Create a test with expect statements using the variables provided.
 
-const secretCodeWord1 = "Lackadaisical"
-// Expected output: "L4ck4d41s1c4l"
-const secretCodeWord2 = "Gobbledygook"
-// Expected output: "G0bbl3dyg00k"
-const secretCodeWord3 = "Eccentric"
-// Expected output: "3cc3ntr1c"
+describe("codedMessage", () => {
+    it("takes in a string and returns a string with a coded message", () => {
+        expect(codedMessage(secretCodeWord1)).toStrictEqual("L4ck4d41s1c4l")
+        const secretCodeWord1 = "Lackadaisical"
+        // Expected output: "L4ck4d41s1c4l"
+        expect(codedMessage(secretCodeWord2)).toStrictEqual("G0bbl3dyg00k")
+        const secretCodeWord2 = "Gobbledygook"
+        // Expected output: "G0bbl3dyg00k"
+        expect(codedMessage(secretCodeWord3)).toStrictEqual("3cc3ntr1c")
+         const secretCodeWord3 = "Eccentric"
+        // Expected output: "3cc3ntr1c"
+    })
+})
+
+// FAIL  ./code-challenges.test.js
+// codedMessage
+//   ✕ takes in a string and returns a string with a coded message
+
+// ● codedMessage › takes in a string and returns a string with a coded message
+
+//   ReferenceError: codedMessage is not defined
 
 // b) Create the function that makes the test pass.
 
+ const codedMessage = (string) =>  {
+ return string.replace(/a/ig, "4").replace(/e/ig, "3").replace(/i/ig, "1").replace(/o/ig, "0")
```

Comment:
Great research and implementation of `.replace` and regex! Nice job chaining them back to back as well to keep it a one liner. Make sure you are paying attention to indentation